pages/index.tsx

```diff
@@ -1,19 +1,25 @@
 import type { NextPage } from "next";
-import useSwr from "swr";
+import { useEffect, useState } from "react";
 import PackageList from "../components/PackageList";
 
-const fetcher = (url: string) => fetch(url).then((res) => res.json())
-
 const Home: NextPage = () => {
+  const [data, setData] = useState(null);
+  const [isLoading, setLoading] = useState(false);
 
-  const { data, error } = useSwr("/api/package-reader",
-    fetcher
-  )
+  useEffect(() => {
+    setLoading(true);
+    fetch("/api/package-reader")
+      .then((data) => data.json())
+      .then((data) => {
+        setData(data);
+        setLoading(false);
+      });
+  }, []);
```

Comment:
Yup, this is almost exactly how I would use the `useEffect` :+1:

Some smaller comments: The `useState` should be typed, for additional type safety, and I'd use a `finally` (I _think_ `fetch` will throw an error if you get a non-200 status error from the backend). Soooo:

```typescript
// in PackageList.tsx
type Item = { name: string; version: string };
type PackageListProps = {
  items: Item[];
};

// then, in this file...
import PackageList, { Item } from "../components/PackageList";
//...
const [data, setData] = useState<Item[]>([])
//...
fetch("/api/package-reader")
  .then((data) => data.json())
  .then((data) => setData(data)) // could even be written just `.then(setData)`, YAY for functional programming! :D
  .finally(() => setLoading(false))
});
```

Reply:
👍 Agree, code becomes much cleaner this way. Suggested changes already applied.